src/sql/statement.rs

```diff
@@ -1789,7 +1790,11 @@ pub struct StatementContext<'a> {
 }
 
 impl<'a> StatementContext<'a> {
-    pub fn allocate_name(&self, name: PartialName) -> FullName {
+    pub fn allocate_name(&self, name: PartialName, temporary: bool) -> FullName {
```

Comment:
In general I try very hard to avoid `bool` parameters in method signatures because it makes the call sites very hard to read—especially since 95% of the callsites want to pass `false` here.  In this case I think you've actually got a very nice API already between `allocate_name` and `allocate_temporary_name`—I would recommend just hoisting the `if temporary` check out to the one place you _do_ want to create a temporary name.